frontend/mainwindow.cpp

```diff
@@ -62,9 +93,13 @@ void MainWindow::fillInstruments() {
   foreach (const QString instrument, instruments) {
     ui->instrumentsBox->addItem(instrument);
   }
+  connect(ui->instrumentsBox, SIGNAL(currentTextChanged(const QString)), this,
+          SLOT(instrumentsBoxChange(const QString)));
 }
 
-void MainWindow::on_instrumentsBox_currentTextChanged(const QString &arg1) {
+void MainWindow::instrumentsBoxChange(const QString &arg1) {
```

Comment:
Why not just implement the `private slot` `on_instrumentsBox_currentTextChanged(const QString)`?

Reply:
This was from a previous solution to preventing unwanted signal firing. Have now updated and simplified